components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -82,7 +82,11 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         const id = getWindowId(tab.windowId)
         const publishers: Record<string, RewardsExtension.Publisher> = state.publishers
         const publisher = publishers[id]
-        chrome.braveRewards.getPublisherData(tab.windowId, tab.url, tab.favIconUrl || '', payload.publisherBlob || '')
+
+        if ((payload.onlyDiff && publisher && publisher.tabUrl !== tab.url) || !payload.onlyDiff) {
```

Comment:
This seems to be causing problems with getting media publishers in panel

Reply:
this is working as expected, fix for the panel not working https://github.com/brave/brave-core/pull/2178